src/vite/plugin.tsx

```diff
@@ -91,13 +96,54 @@ export const remixDevTools: (args?:RemixViteConfig) => Plugin[] = (args) => {
           }))
         });
 
-        if(!server.config.isProduction){
-           const { exec } = await import ("node:child_process") 
-          server.hot.on("open-source", (data) => { 
-          const source = data.data.source; 
-          const line = data.data.line;
-          exec(`code -g "${normalizePath(source)}:${line}"`)
-          })
+        if (!server.config.isProduction) {
+          const { exec } = await import("node:child_process");
+
+          server.hot.on("open-source", ({ data }: OpenSourceData) => {
+            const { source, line, routeID } = data;
+            const lineNum = line ? `:${line}` : "";
+
+            if (source) {
+              exec(`code -g "${normalizePath(source)}${lineNum}"`);
+              return;
+            }
+
+            if (!source && routeID) {
+              const routePath = path.join(remixDir, routeID);
+              const checkedPath = checkPath(routePath);
+
+              if (!checkedPath) return;
+              const { type, validPath } = checkedPath;
+
+              const reactExtensions = ["tsx", "jsx"];
+              const allExtensions = ["ts", "js", ...reactExtensions];
+              const isRoot = routeID === "root";
+
+              if (isRoot) {
+                if (!fs.existsSync(remixDir)) return;
+                const filesInRemixPath = fs.readdirSync(remixDir);
+                const rootFile = filesInRemixPath.find((file) => reactExtensions.some((ext) => file === `root.${ext}`));
+
+                rootFile && exec(`code -g "${path.join(remixDir, rootFile)}${lineNum}"`);
```

Comment:
extract this into a function called openInVsCode or something because here the path is not normalized and above it is,
something like:
```ts
function openInVsCode(path: string){
  exec(`code -g "${normalizePath(path)}"`˙)
}
```